NEWS

```diff
@@ -4,12 +4,100 @@ Noteworthy changes in release a.b
-
 * In case a PG header line has multiple ID tags supplied by other applications,
   the header API now selects the first one encountered as the identifying tag
   and issues a warning when detecting subsequent ID tags.
+  (#1256; fixed samtools/samtools#1393)
+
+* VCF header reading function (vcf_hdr_read) no longer tries to download a
+  remote index file by default.
+  (#1266; fixes #380)
+  
+* Transparently treat FASTQ as unmapped data as if it was SAM, BAM or CRAM.
+  (#1156)
+
+* Added GCP requester pays bucket access.  Thanks to @indraniel.
+  (#1255)
+  
+* Made mpileup's overlap removal choose a random sequence.
+  (#1273; fixes samtools/bcftools#1459)
+  
+* Now permits platform specific BAQ parameters.  This also selects long-read
+  parameters for read lengths bigger than 1kb.  This helps bcftools mpileup call
+  SNPs on PacBio CCS reads.
+  (#1275)
+  
+* Improved bcf_remove_allele_set.  This fixes a bug that stopped iteration over
+  alleles prematurely, marks removed alleles as 'missing' and does automatic
+  lazy unpacking.
+  (#1288; fixes #1259)
+  
+* [CRAM] Improved compression metrics for unsorted files.  This improves the
+  choice of codecs when handling unsorted data.
+  (#1291)
+  
+* Now initialisea the missing entries of the linear index in reverse order.
+  Populates empty linear index entries from the right rather from the left.
+  The new hts_bin_level function computes the index level of a bin.
+  Thanks to @carsonh for reporting the issue.
+  (#1286; fixes #486)
+  
+* Related to the above, the new method, hts_idx_nseq, now returns the total
+  number of contigs from an index.
+  (#1295 and #1299)
+   
+* Added bracket handling to bcf_hdr_parse_line.  Thanks to Alberto Casas Orrtiz.
+
+Build changes
+-------------
+
+These are compiler, configuration and makefile based changes.
+
+* Added a curl/curl.h check to configure and improved INSTALL documentation on
+  build options.  Thanks to Melanie Kirsche and John Marshall.
+  (#1265; fixes #1261)
+
+* Some fixes to address GCC 11.1 warnings.
+  (#1280, #1284, #1285; fixes #1283)
+
+* Supports building HTSlib in a separate directory. Thanks to John Marshall.
+  (#1277; fixes #231)
+
+* Supports building HTSlib on MinGW 32-bit environments. Thanks to
+  John Marshall.
+  (#1301)
+
+Bug fixes
+---------
+
+* Removed compressBound assertions on opening bgzf files.  Thanks to 
+  Gurt Hulselmans for reporting the issue.
+  (#1258; fixed #1257) 
+
+* Duplicate sample name error message for a VCF file now only displays the
+  duplicated name rather the entire same name list.
+  (#1262; fixes samtools/bcftools#1451)
+  
+* Fix to make samtools cat work on CRAMs again.
+  (#1276; fixes samtools/samtools#1420)
+  
+* Fix for a double memory free in SAM header creation.  Thanks to @ihsineme.
+  (#1274)
+  
+* Prevent assert in bcf_sr_set_regions.  Thanks to Dr K D Murray.
+  (#1270)
+  
+* Fixed the update of the minimum offset when unmapped reads are present.  This
```

Comment:
This is a serious bug that should be more clearly described, placed at the top of the list, and probably noted in samtools's _NEWS_ as well. Perhaps something like

> Fixed `hts_itr_query()` et al region queries: fixed bug introduced in HTSlib 1.12, which lead to iterators producing very few reads for some queries (especially for larger target regions) when unmapped reads were present. (HTSlib 1.11 had a related problem in which iterators would omit a few unmapped reads that should have been produced; cf #1142.)

This has probably been reported multiple times: #1279 as listed (first and best researched, so the credit for Daniel is certainly deserved), but also samtools/samtools#1451, possibly #1290, and probably samtools/samtools#1457.

Over in samtools, this could be described in terms of region queries instead:

> Fixed `samtools view FILE REGION` and `mpileup -r REGION` etc region queries: fixed bug introduced in 1.12, which lead to region queries producing very few reads for […]

Reply:
Changed. Thanks!

Reply:
Thanks for applying. It'd probably be worth listing the additional bug report numbers, and I'm afraid it should be `s/lead/led/` — I'm pretty good at mixing that one up :smile:

Reply:
Perhaps the other bug reports should be mentioned in the SAMtools news, as they are related to running samtools commands.

Reply:
I guess where HTSlib bugs noticeably affect particular samtools outputs, it can be worth mentioning them in the samtools context. Alas, who knows whether anyone actually reads the release notes for either samtools or htslib… 🤷